src/jit/gentree.cpp

```diff
@@ -6246,15 +6246,310 @@ bool GenTree::TryGetUse(GenTree* def, GenTree*** use)
     assert(def != nullptr);
     assert(use != nullptr);
 
-    for (GenTree** useEdge : UseEdges())
+    switch (OperGet())
+    {
+        // Leaf nodes
```

Comment:
Is the switch more efficient than checking for `GTK_LEAF`, `GTK_UNOP`, etc? That would certainly be cleaner, and then you could have cases for any nodes that require special handling.